app/src/main/kotlin/org/kotlinlsp/index/queries/Completions.kt

```diff
@@ -1,9 +1,22 @@
 package org.kotlinlsp.index.queries
 
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.decodeFromByteArray
+import kotlinx.serialization.protobuf.ProtoBuf
 import org.kotlinlsp.index.Index
+import org.kotlinlsp.index.db.Declaration
 
-fun Index.getCompletions(prefix: String): List<String> = query {
+@OptIn(ExperimentalSerializationApi::class)
+fun Index.getCompletions(prefix: String, thisRef: String, receiver: String): List<Declaration> = query {
```

Comment:
I'd return a `Sequence<Declaration>` here as we are doing more operations with this in the autocomplete action, and call `toList()` at the end of it